hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1685,10 +1696,11 @@ private static Comparable<?> coerceToComparable(Schema schema, Object val) {
   private static boolean isColumnTypeSupported(Schema schema, Option<HoodieRecordType> recordType) {
     // if record type is set and if its AVRO, MAP is unsupported.
     if (recordType.isPresent() && recordType.get() == HoodieRecordType.AVRO) {
-      return schema.getType() != Schema.Type.MAP;
+      return (schema.getType() != Schema.Type.RECORD && schema.getType() != Schema.Type.ARRAY && schema.getType() != Schema.Type.MAP);
     }
     // if record Type is not set or if recordType is SPARK then we cannot compare RECORD and ARRAY types in addition to MAP type
-    return schema.getType() != Schema.Type.RECORD && schema.getType() != Schema.Type.ARRAY && schema.getType() != Schema.Type.MAP;
+    return schema.getType() != Schema.Type.RECORD && schema.getType() != Schema.Type.ARRAY && schema.getType() != Schema.Type.MAP
```

Comment:
HoodieTableMetadataIUtil, HoodieAvroUtils (schema managing code snippets) are all part of hudi-common. So, some of the spark schema types are supported w/ col stats yet. for AVRO record and avro schema, we are find, just that for spark record type and spark schema (structType), we may not have support for some data types. HUDI-8585 will help fix those.